crates/nu-command/src/filesystem/cd.rs

```diff
@@ -271,7 +271,14 @@ fn have_permission(dir: impl AsRef<Path>) -> PermissionResult<'static> {
     }
 }
 
-#[cfg(unix)]
+#[cfg(any(target_os = "linux", target_os = "android"))]
+fn any_group(_current_user_gid: gid_t, owner_group: u32) -> bool {
+    use crate::filesystem::util::users;
+    let Some(user_groups) = users::current_user_groups() else {return false};
+    user_groups.iter().any(|gid| gid.as_raw() == owner_group)
+}
+
+#[cfg(all(unix, not(any(target_os = "linux", target_os = "android"))))]
```

Comment:
So this can be `[cfg(all(unix, not(target_os = "android")))]`

Reply:
as above, this more efficient way of checking group for linux can done using above implementation so